File: csharp/log-levels/LogLevels.cs
using System;
using System.Text.RegularExpressions;

static class LogLine
{
    static string[] logLevels = { "ERROR", "INFO", "WARNING" };

    public static string Message(string logLine)
    {
        foreach (string logLevel in logLevels)
        {
            if (isAtLogLevel(logLine, logLevel))
            {
                return extractLogFromLine(logLine, formatLogLevel(logLevel));
            }
        }

        return logLine;
    }

    private static bool isAtLogLevel(string logLine, string logLevel)
    {
        return logLine.IndexOf(formatLogLevel(logLevel)) == 0;
    }

    private static string extractLogFromLine(string logLine, string level)
    {
        return logLine.Substring(level.Length).Trim();
    }

    public static string LogLevel(string logLine)
    {
        foreach (string logLevel in logLevels)
        {
            if (isAtLogLevel(logLine, logLevel))
            {
                return logLevel.ToLower();
            }
        }

        return logLine;
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }

    private static string formatLogLevel(string logLevel)
    {
        return "[" + logLevel + "]:";
    }
}

File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {

    public static String message(String logLine) {
        String message = logLine.split(":")[1];

        return message.trim();
    }

    public static String logLevel(String logLine) {
        String level = logLine.split(":")[0];

        return level.trim().toLowerCase().replace("[", "").replace("]", "");
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}
